### PR TITLE
feat: improve dynamic link contracts' tests and add them to github workflow

### PR DESCRIPTION
**Pull request by loloicci**
# Description
This PR improves dynamic link contracts' tests and adds them to github workflow.

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/cosmwasm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly. (not needed)
- [x] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

**Comment by zemyblue** [comment]
Please add the changelog about this PR if necessary.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -699,3 +797,16 @@ jobs:
...
+      #   run: |
+      #       mkdir -p target/wasm32-unknown-unknown/release
+      #       touch target/wasm32-unknown-unknown/release/dynamic_callee_contract.wasm
+      #       cargo clippy --tests -- -D warnings
+      # - name: Clippy linting on dynamic-caller-contract
+      #   working-directory: ${{env.contracts}}/dynamic-caller-contract
+      #   run: |
+      #       mkdir -p target/wasm32-unknown-unknown/release
+      #       touch target/wasm32-unknown-unknown/release/dynamic_caller_contract.wasm
+      #       cargo clippy --tests -- -D warnings
```
[the remark this answers is not in the document]
Comment (loloicci) [reply .github/workflows/tests.yml:812]: > As far as know the version we are using is 1.57.0, so I think it is not problem through the comment.
You are right. But, other tests issues error with 1.57.0 like https://github.com/line/cosmwasm/runs/7234121824?check_suite_focus=true. So, I will add them when it is merged into the main branch.

**Comment by loloicci** [comment]
@zemyblue 

> Please add the changelog about this PR if necessary.

In our repository, CHANGELOG will be edited when the version bumps. So I think it is unnecessary. Isn't it right?

File: .github/workflows/tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zemyblue) [reply .github/workflows/tests.yml:812]: It means you'll add the another PR after merging this PR?

**Comment by zemyblue** [comment]
> @zemyblue
> 
> > Please add the changelog about this PR if necessary.
> 
> In our repository, CHANGELOG will be edited when the version bumps. So I think it is unnecessary. Isn't it right?

OK

**Review by shiki-tak — APPROVED** [review]
LGTM

**Review by zemyblue — APPROVED** [review]
LGTM

File: .github/workflows/tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (loloicci) [reply .github/workflows/tests.yml]: Yes, before merging into the main.

**Pull request merged**